notebook/tree/handlers.py

```diff
@@ -49,6 +50,7 @@ def get(self, path=''):
                 notebook_path=path,
                 breadcrumbs=breadcrumbs,
                 terminals_available=self.settings['terminals_available'],
+                server_root=os.path.basename(os.path.normpath(self.settings['server_root_dir'])),
```

Comment:
@gnestor  I think the idea was for it to still be an absolute path, but collapse home to `~`. So instead of basename, call:

```python
root_dir = self.contents_manager.root_dir
home = os.path.expanduser('~')
if root_dir.startswith(home + os.path.sep):
    # collapse $HOME to ~
    root_dir = '~' + root_dir[len(home):]
```

This can be done once when defining the `server_root_dir` above in `notebookapp.py`, rather than here every time. Here, we can use `server_root=self.settings['server_root']`, assuming it already has the right form.